storm-core/src/jvm/org/apache/storm/StormTimer.java

```diff
@@ -0,0 +1,231 @@
+                        // sleeping time, to limit the response time
+                        // for detecting any new event within 1 secs.
+                        Time.sleep(Math.min(1000, (queueEntry.endTimeMs - Time.currentTimeMillis())));
+                    } else {
+                        // Otherwise poll to see if any new event
+                        // was scheduled. This is, in essence, the
+                        // response time for detecting any new event
+                        // schedulings when there are no scheduled
+                        // events.
+                        Time.sleep(1000);
+                    }
+                } catch (Throwable t) {
+                    if (!(Utils.exceptionCauseIsInstanceOf(InterruptedException.class, t))) {
+                        this.onKill.run(t);
+                        this.setActive(false);
+                        throw new RuntimeException(t);
+                    }
+                }
+            }
+            this.cancelNotifier.release();
+        }
+
+        public void setOnKillFunc(TimerFunc onKill) {
+            this.onKill = onKill;
+        }
+
+        public void setActive(boolean flag) {
+            this.active.set(flag);
+        }
+
+        public boolean isActive() {
+            return this.active.get();
+        }
+
+        public void add(QueueEntry queueEntry) {
+            this.queue.add(queueEntry);
+        }
+    }
+
+    public static StormTimerTask mkTimer(String name, TimerFunc onKill) {
+        if (onKill == null) {
+            throw new RuntimeException("onKill func is null!");
+        }
+        StormTimerTask task  = new StormTimerTask();
+        if (name == null) {
+            task.setName("timer");
+        } else {
+            task.setName(name);
+        }
+        task.setOnKillFunc(onKill);
+        task.setActive(true);
+
+        task.setDaemon(true);
+        task.setPriority(Thread.MAX_PRIORITY);
+        task.start();
+        return task;
+    }
+    public static void schedule(StormTimerTask task, int delaySecs, TimerFunc afn, boolean checkActive, int jitterMs) {
+        if (task == null) {
+            throw new RuntimeException("task is null!");
+        }
+        if (afn == null) {
+            throw new RuntimeException("function to schedule is null!");
+        }
+        String id = Utils.uuid();
+        long endTimeMs = Time.currentTimeMillis() + Time.secsToMillisLong(delaySecs);
+        if (jitterMs > 0) {
+            endTimeMs = task.random.nextInt(jitterMs) + endTimeMs;
+        }
+        synchronized (task.lock) {
+            task.add(new QueueEntry(endTimeMs, afn, id));
+        }
+    }
+    public static void schedule(StormTimerTask task, int delaySecs, TimerFunc afn) {
+        schedule(task, delaySecs, afn, true, 0);
+    }
+
+    public static void scheduleRecurring(final StormTimerTask task, int delaySecs, final int recurSecs, final TimerFunc afn) {
```

Comment:
can you keep the order of arguments consistent with above scheduleFunction? 
`
StormTimerTask task, int delaySecs, TimerFunc afn, final int recurSecs
`